mmv1/third_party/tgc_next/pkg/tfplan2cai/converters/services/compute/compute_instance.go

```diff
@@ -9,16 +9,1429 @@ import (
 	"github.com/GoogleCloudPlatform/terraform-google-conversion/v6/pkg/tfplan2cai/converters/cai"
 
 	"github.com/hashicorp/terraform-provider-google-beta/google-beta/tpgresource"
 	transport_tpg "github.com/hashicorp/terraform-provider-google-beta/google-beta/transport"
+	"github.com/hashicorp/terraform-provider-google-beta/google-beta/verify"
 )
 
 const ComputeInstanceAssetType string = "compute.googleapis.com/Instance"
 const ComputeDiskAssetType string = "compute.googleapis.com/Disk"
 
+var (
+	advancedMachineFeaturesKeys = []string{
+		"advanced_machine_features.0.enable_nested_virtualization",
+		"advanced_machine_features.0.threads_per_core",
+		"advanced_machine_features.0.turbo_mode",
+		"advanced_machine_features.0.visible_core_count",
+		"advanced_machine_features.0.performance_monitoring_unit",
+		"advanced_machine_features.0.enable_uefi_networking",
+	}
+
+	bootDiskKeys = []string{
+		"boot_disk.0.guest_os_features",
+		"boot_disk.0.auto_delete",
+		"boot_disk.0.device_name",
+		"boot_disk.0.disk_encryption_key_raw",
+		"boot_disk.0.kms_key_self_link",
+		"boot_disk.0.disk_encryption_key_rsa",
+		"boot_disk.0.disk_encryption_service_account",
+		"boot_disk.0.initialize_params",
+		"boot_disk.0.mode",
+		"boot_disk.0.source",
+	}
+
+	initializeParamsKeys = []string{
+		"boot_disk.0.initialize_params.0.size",
+		"boot_disk.0.initialize_params.0.type",
+		"boot_disk.0.initialize_params.0.image",
+		"boot_disk.0.initialize_params.0.labels",
+		"boot_disk.0.initialize_params.0.resource_manager_tags",
+		"boot_disk.0.initialize_params.0.provisioned_iops",
+		"boot_disk.0.initialize_params.0.provisioned_throughput",
+		"boot_disk.0.initialize_params.0.enable_confidential_compute",
+		"boot_disk.0.initialize_params.0.source_image_encryption_key",
+		"boot_disk.0.initialize_params.0.snapshot",
+		"boot_disk.0.initialize_params.0.source_snapshot_encryption_key",
+		"boot_disk.0.initialize_params.0.storage_pool",
+		"boot_disk.0.initialize_params.0.resource_policies",
+		"boot_disk.0.initialize_params.0.architecture",
+	}
+
+	schedulingKeys = []string{
+		"scheduling.0.on_host_maintenance",
+		"scheduling.0.automatic_restart",
+		"scheduling.0.preemptible",
+		"scheduling.0.node_affinities",
+		"scheduling.0.min_node_cpus",
+		"scheduling.0.provisioning_model",
+		"scheduling.0.instance_termination_action",
+		"scheduling.0.termination_time",
+		"scheduling.0.availability_domain",
+		"scheduling.0.max_run_duration",
+		"scheduling.0.on_instance_stop_action",
+		"scheduling.0.maintenance_interval",
+		"scheduling.0.host_error_timeout_seconds",
+		"scheduling.0.graceful_shutdown",
+		"scheduling.0.local_ssd_recovery_timeout",
+	}
+
+	shieldedInstanceConfigKeys = []string{
+		"shielded_instance_config.0.enable_secure_boot",
+		"shielded_instance_config.0.enable_vtpm",
+		"shielded_instance_config.0.enable_integrity_monitoring",
+	}
+)
+
+func ResourceComputeInstance() *schema.Resource {
```

Comment:
How will we make sure that the schemas in TGC stay in sync with the schemas in the provider?

Reply:
For handwritten resources, the schemas in TGC will be updated when new fields are supported for conversions. There is no automatic sync between schemas in TGC  and the schemas int he provider for handwritten resources.

Reply:
fair enough, things that can't be handled don't need to be here. SGTM.